frontend/tests/unit/services/transactions/utils.spec.ts

```diff
@@ -2,18 +2,88 @@ import * as providers from '@ethersproject/providers';
 import * as ethers from 'ethers';
 
 import {
+  CONFIRMATION_TIME_BLOCKS,
   getCurrentBlockNumber,
   getJsonRpcProvider,
   getLatestBlock,
   getReadOnlyContract,
   getReadWriteContract,
+  getSafeEventHandler,
 } from '@/services/transactions/utils';
 import { getRandomEthereumAddress, getRandomUrl } from '~/utils/data_generators';
+import { MockedEvent, MockedTransactionReceipt } from '~/utils/mocks/ethers';
 
 vi.mock('@ethersproject/providers');
 vi.mock('ethers');
 
 describe('utils', () => {
+  describe('getSafeEventHandler()', () => {
+    it('returns a re-org safe version of the passed handler', () => {
+      const provider = new providers.JsonRpcProvider();
+      provider.getNetwork = vi.fn().mockResolvedValue({ chainId: 1 });
+      const handler = vi.fn();
+
+      const safeHandler = getSafeEventHandler(handler, provider);
+      expect(safeHandler).toBeTypeOf('function');
+    });
+    describe('returned safe event handler', () => {
+      it('on call waits until event transaction has been finalized before running the related action', async () => {
+        const confirmationTimeBlocks = 5;
+        const chainId = 100;
+        const eventTxHash = '0x123';
+        const event = new MockedEvent({ removed: 0, transactionHash: eventTxHash });
+        const handler = vi.fn();
+
+        CONFIRMATION_TIME_BLOCKS[chainId] = confirmationTimeBlocks;
+
+        const provider = new providers.JsonRpcProvider();
+        provider.getNetwork = vi.fn().mockResolvedValue({ chainId });
+        provider.waitForTransaction = vi
+          .fn()
+          .mockResolvedValue(new MockedTransactionReceipt({ status: 1 }));
+
+        const safeHandler = getSafeEventHandler(handler, provider);
+        await safeHandler(event);
```

Comment:
The handler isn't async. The whole test function doesn't have to be.

Reply:
Actually, the handler isn't the one we are awaiting for but instead the wrapped function around the handler which is async. 
If i remove the await tests will fail.

Reply:
woops, sorry was wrong about this. My IDE marked it as not awaitable. But the reason is just that the return type of the `getSafeEventHandler` is not accurate. The `Listener` type does not return a Promise. The return type should instead be something like: `(...args: Array<unknown>) => Promise<void>`